### PR TITLE
fix(scss): scss config vars, missing imports and hotfixes

### PR DESCRIPTION
**Pull request by hellobrian**
## Overview

resubmission of #72 
related https://github.com/carbon-design-system/carbon-components/issues/71
resolves https://github.ibm.com/Bluemix/bluemix-components-react/issues/610

### Add `_css--typography.scss`

This is all of our `h1`, `h2`, `h3` etc styles that used to be part of `_css--body.scss`.
This new SCSS file is controlled by a **new variable**: `$css--typography: true !default;`

### New defaults for global SCSS variables

```scss
$css--font-face: false !default;
$css--helpers: true !default;
$css--body: true !default;
$css--reset: true !default;
$css--typography: true !default;
```

All variables default to `true` except for `$css--font-face: false !default;`.

### New global SCSS variable conditionals

```scss
@if global-variable-exists('css--body') == false or $css--body == true {
  @include css-body;
}
```
Conditionals have changed.

CSS output occurs when:
- if the `$var` **does not** exist, then output CSS.
- if the `$var` equals `true`, then output CSS

The user can opt out of CSS output by setting the variable to `false`.
- `$css--body: false;` will prevent CSS output. 

### 🔥 Fixes Fixes Fixes (so many fixes, 🙏 🇨🇦)

- Add missing `@import` statements (tested for importing single components w/o vars).
- Remove `<use>` and inline SVG icons in HTML files
- Change SCSS for various visual fixes, see comments for details.

### CODE REVIEW

File: src/globals/scss/_css--body.scss [hunk 1]
```diff
@@ -4,72 +4,8 @@
...
-
-  a {
-    @include reset;
-    color: $brand-01;
-  }
-}
-
 @mixin css-body {
-  .bx--body {
+  body {
```
Comment (hellobrian) [inline src/globals/scss/_css--body.scss]: Setting `$css--body: true` will output these `body` styles to CSS. 
This is the default.

File: src/globals/scss/_css--helpers.scss [hunk 2]
```diff
@@ -13,6 +13,14 @@
     visibility: visible;
     white-space: nowrap;
   }
+
+  .bx--body {
```
Comment (hellobrian) [inline src/globals/scss/_css--helpers.scss]: `.bx--body` class gets applied when `$css--helpers: true`, which is the same styles as `body` from `_css--body.scss`

File: src/globals/scss/styles.scss [hunk 3]
```diff
@@ -6,6 +6,7 @@ $css--font-face: false !default;
 $css--helpers: true !default;
 $css--body: true !default;
 $css--reset: false !default;
+$css--typography: true !default;
```
Comment (hellobrian) [inline src/globals/scss/styles.scss]: `$css--typography: true` will output CSS from `_css--typography.scss` and this is default.

File: src/globals/scss/_css--typography.scss [hunk 4]
```diff
@@ -0,0 +1,76 @@
+@import 'colors';
+@import 'vars';
+@import 'mixins';
+@import 'typography';
+@import 'reset';
```
Comment (tw15egan) [inline src/globals/scss/_css--typography.scss]: I think this should be `css--reset`

File: src/globals/scss/_css--helpers.scss [hunk 5]
```diff
@@ -1,3 +1,5 @@
+@import 'reset';
```
Comment (tw15egan) [inline src/globals/scss/_css--helpers.scss]: I think this should be `css--reset`

File: src/globals/scss/_css--body.scss [hunk 6]
```diff
@@ -2,74 +2,11 @@
 @import 'vars';
 @import 'mixins';
 @import 'typography';
+@import 'reset';
```
Comment (tw15egan) [inline src/globals/scss/_css--body.scss]: I think this should be `css--reset`

**Review by tw15egan — CHANGES_REQUESTED** [review]
👍 ✅ ❗️

File: src/globals/scss/_css--typography.scss (hunk 4 above)
Comment (hellobrian) [reply src/globals/scss/_css--typography.scss]: blargh! sorry, thanks for pointing this out haha

File: src/components/accordion/_accordion.scss [hunk 7]
```diff
@@ -38,8 +39,8 @@
 
   .bx--accordion__arrow {
     transition: all $transition--base $bx--standard-easing;
-    height: .75rem;
-    width: .75rem;
+    height: 1.25rem;
+    width: 1.25rem;
```
Comment (hellobrian) [inline src/components/accordion/_accordion.scss:43]: Change: `padding` was affecting size of accordion arrow icons making them appear smaller than normal.

File: src/components/card/_card.scss [hunk 8]
```diff
@@ -85,17 +81,21 @@
     @include text-overflow(rem(180px));
     font-weight: 400;
     margin: 0;
+    line-height: 1.2;
```
Comment (hellobrian) [inline src/components/card/_card.scss:84]: Change: new styles here fix unwanted scrollbar from appearing

image

☝️ with unwanted scrollbar

File: src/components/file-uploader/_file-uploader.scss [hunk 9]
```diff
@@ -58,20 +55,19 @@
   .bx--file-filename {
     @include font-size('12');
     @include text-overflow(100%);
+    display: inline-flex;
+    align-items: center;
     color: $text-01;
     margin-right: 1rem;
+    height: 1.875rem;
```
Comment (hellobrian) [inline src/components/file-uploader/_file-uploader.scss:62]: Change: styles added here fix unwanted scrollbar and prevents it from appearing

File: src/components/notification/_inline-notification.scss [hunk 10]
```diff
@@ -17,6 +17,7 @@
     @include font-smoothing;
     @include font-size('14');
     display: flex;
+    justify-content: space-between;
```
Comment (hellobrian) [inline src/components/notification/_inline-notification.scss:20]: Change: fixes missing space between inline-notification text and close button.

image

☝️  missing space

File: src/components/progress-indicator/_progress-indicator.scss [hunk 11]
```diff
@@ -25,7 +25,7 @@
   .bx--progress-line {
     position: absolute;
     top: .625rem;
-    right: 99.5%;
+    right: 100%;
```
Comment (hellobrian) [inline src/components/progress-indicator/_progress-indicator.scss:28]: Change: fixes gap between line and indicators.

File: src/components/search/_search.scss [hunk 12]
```diff
@@ -101,6 +101,7 @@
     fill: $brand-01;
     height: 1rem;
     width: 1rem;
+    vertical-align: middle;
```
Comment (hellobrian) [inline src/components/search/_search.scss]: Need to add more styles

```css

    position: relative;
    top: -2px;
```

File: src/components/toolbar/_toolbar.scss [hunk 13]
```diff
@@ -93,6 +93,10 @@ $css--helpers: true;
     border: 0;
     height: rem(32px);
     width: rem(32px);
+
+    &:focus {
+      @include focus-outline;
```
Comment (hellobrian) [inline src/components/toolbar/_toolbar.scss:98]: Change: fixes missing focus outline on toolbar search button

File: src/components/tooltip/_tooltip.scss [hunk 14]
```diff
@@ -12,6 +13,7 @@
   }
 
   .bx--tooltip__trigger {
+    @include helvetica;
```
Comment (hellobrian) [inline src/components/tooltip/_tooltip.scss:16]: Change: adds Helvetica font-family to tooltip triggers

File: src/components/search/_search.scss [hunk 15]
```diff
@@ -94,9 +94,14 @@
...
     background-color: $ui-01;
     position: relative;
+    padding: 0;
   }
 
   .bx--search-button svg {
+    position: relative;
+    top: -1px;
+    box-sizing: border-box;
+    vertical-align: middle;
```
Comment (hellobrian) [inline src/components/search/_search.scss:104]: Change: centers icon in buttons

**Review by tw15egan — APPROVED** [review]

**Comment by hellobrian** [comment]
@chrisdhanaraj Went through the components one more time and made sure all components that need `_css--typography.scss` are now importing it.

**Comment by hellobrian** [comment]
@chrisdhanaraj says: 👍 ✅

**Comment by chrisdhanaraj** [comment]
👍 ✅

**Pull request merged**